### ISSUE TITLE
Measurable functions definition

### ISSUE THREAD

**Issue by qucchia** [vEnhance/napkin#257]
In the definition of measurable function, it is not clear what an "open set" S means. Is it implied that $(Y, \mathcal{B})$ is a Borel $\sigma$-algebra? If so I think it should be made more explicit somewhere.

![image](https://github.com/user-attachments/assets/ff7f6e32-4f48-4d3f-b35b-0d185300fe2f)

**Comment by user202729** [comment]
Actually… well… oops.

I mean, that's the usual definition when (Y, ℬ) is a _topological_ space, which is the common case e.g. X ⊆ ℝ^n and Y = ℝ, so your f is something you want to integrate.

When (Y, ℬ) just have a _measure_ without a topology…

(I haven't met a case where you need that definition of measurable function either.)

**Comment by vEnhance** [comment]
So do y'all think it makes sense to just require (Y, B) to be Borel sigma-algebra or revert to the definition beforehand? My impression is that it shouldn't matter much either way (if I'm wrong, correct me on this).

**Comment by user202729** [comment]
I can't say I have had enough exposure to map between two measurable spaces to say this (all measurable function I've encountered is from some measure space to ℝ, which is then to be integrated), but I feel there are two distinct things:

* measurable functions i.e. morphism of measure spaces (X, 𝒜) → (Y, ℬ) (the only sensible definition in this case is to lift measurable set to measurable set, then it satisfies well-behaved rules such as closure under composition)
* function (X, 𝒜) → ℝ that is meant to be integrated over a measurable set. In this case the only sensible definition is that it lifts open sets to measurable sets.

I feel the abuse of terminology is that the two are conflated (since they coincide when the right hand side uses Borel algebra). Though as I mentioned, I don't know if there's some deeper underlying reason.

*Maybe* this is the reason for "From now on, we assume the Borel measure" part. (I don't know why either. My classes use Lebesgue measure.)

**Comment by user202729** [comment]
tl;dr

* Option 1. Revert to original definition, add "for our purpose here, for functions f: X → ℝ and we want to see if it's integrable, assume ℝ has Borel algebra".

  Advantage: then you can compose two measurable functions.

  Disadvantage: I think the material in the book is not sufficient to motivate why we need maps between two measure space instead of map from measure space to topological space.

* Option 2. change to "let X be measure space and Y be topological space, then f is measurable if it lift open set to measurable set".

  Advantage: Easier to motivate e.g. U ⊆ X measurable set then characteristic(U): X → ℝ is measurable function.

  Disadvantage: see option 1's advantage.

* Option 3: change to "let X be measure space and Y be ℝ"

  This is just (option 2) × 2 - (option 1) sort of. In the spirit of "just show a prototypical example" it works.

**Comment by vEnhance** [comment]
Sorry for dropping this for way more months than I realized. I created PR #270 which implements Option 1 --- I think being able to compose measurable functions is okay as motivation, since it specializes correctly when Y is topological space.

Putting up as a PR just in case anyone wants to glance at it and make sure I did things right. Will merge in a bit if there's no further comments.

**Comment by user202729** [comment]
> it specializes correctly when Y is topological space

Assuming Borel σ-algebra though.

That said, I've never encountered a case where I need to work with some more exotic measure space, nor have I seen a case where the codomain of a measurable function not ℝⁿ. So I don't know what the motivation is supposed to be either.
